### DISCUSSION TITLE
Don't override slug if already set

### DISCUSSION THREAD

**Discussion by sapkra** [ComfortablyCoding/strapi-plugin-slugify#57] [General]
If you already have published an article with a title and slug and you want to change the title afterwards, it would be very bad if the URL is changing too.

There should be a setting to disable that behavior.

**Comment by ComfortablyCoding** [comment]
By making the title field a reference you are adding it as a dependancy for the slug. Their is no way for the program to know when you want to update the slug or not when the title is updated. 

If you have any suggestions on how to handle this, I am happy to hear them.

P.S. I am moving this to a discussion as it is not an issue/bug.

**Reply by sapkra** [reply]
Isn't it just like adding `if (data[field]) return;` in [this line](https://github.com/ComfortablyCoding/strapi-plugin-slugify/blob/master/server/services/slug-service.js#L20)? Just tried it and it seems to work as expected.

**Reply by ComfortablyCoding** [reply]
Yes it is possible to prevent it from being updated after it has been generated. My point was most users will want it updated when the data changes and their is no way to differentiate between when a user will want it updated and when they want it skipped.

Event if we add a setting for example to indicate to only generate it once, what happens when you create a page with a title that has a typo for example, will you not want the slug to be updated as well? 

I would recommend to use a different field as a reference if title is something you dont want slug to be depended on.

**Reply by sapkra** [reply]
Ok, I will have a look and will possibly provide a PR with a setting/option.

> What happens when you create a page with a title that has a typo for example, will you not want the slug to be updated as well?

No, I don't because when the page URL is changing then and I need to create a redirect from the old to the new one somehow (because of SEO and links from other websites). For that I would need to save the old slugs somehow and this case would be additionally hard to handle for static side generators.
If I really want to change it because it's still in draft mode, then I would be able to empty the slug field, save it and it will generate a new slug from the new title.

**Comment by ComfortablyCoding** [comment]
This was fixed in #68 and should be fixed for creation case once #78 is merged.

The default is to not update, but this behaviour can be changed via the shouldUpdateSlug setting.
